lib/core/src/Cardano/Wallet.hs

```diff
@@ -845,6 +847,27 @@ estimatePaymentFee ctx wid recipients = do
   where
     tl = ctx ^. transactionLayer @t @k
 
+-- | Estimate the transaction fee for joining a stake pool
+estimateJoinStakePoolFee
+    :: forall ctx s t k.
+        ( HasTransactionLayer t k ctx
+        , HasDBLayer s k ctx
+        , HasLogger ctx
+        )
+    => ctx
+    -> WalletId
+    -> ExceptT ErrEstimateJoinStakePoolFee IO Fee
+estimateJoinStakePoolFee ctx wid = do
+    (wal, _, _) <- withExceptT ErrEstimateJoinStakePoolFeeNoSuchWallet $
+        readWallet @ctx @s @k ctx wid
+    sel <- withExceptT ErrEstimateJoinStakePoolSelectForDelegation $
+        selectCoinsForDelegation @ctx @s @t @k ctx wid
+    let bp = blockchainParameters wal
+    let opts = feeOpts tl (computeCertFee 1) (bp ^. #getFeePolicy)
+    return $ estimateFee opts sel
```

Comment:
**[change required]**

Use instead `inputBalance`, `changeBalance` and `outputBalance` to compute the _actual fee_ on a given selection.

**Why?**

There's no need to re-estimate fees whereas they can really be known exactly. This has little importance for Jörmungandr but is _more correct_ and will matter for the Haskell nodes.